### DISCUSSION TITLE
`NuxtError.data` is always a JSON string on SSR

### DISCUSSION THREAD

**Discussion by kabalage** [nuxt/nuxt#27687] [Questions]
I want to customize `error.vue`, and I have some custom data to pass with the error. The recommended way is to set `data` when calling `createError()`. When the error is thrown during SSR, `error.data` becomes a JSON, but it's never parsed on the client side and it stays a string. However if the same error is thrown during client-side rendering, `error.data` is kept as is.

Reproduction: https://stackblitz.com/edit/github-hkoejs

Is this intended or it's a bug? The documentation does not mention this quirk at all.

**Comment by bastienrossi** [comment]
Hi,
I think it's a bug.
From what I understand the error is stored in the payload to be used on the client side when hydrating.
As a result, it's not indicated in the documentation but ‘data’ should be serialisable because if, for example, we put a FetchResponse object it won't serialise correctly.
In my opinion, however, it should deserialise data after retrieving it from the payload.
I'm trying to find the source of the problem and I've noticed this difference between the client and server payload implementations.
packages/nuxt/src/app/plugins/revive-payload.server.ts
```NuxtError: data => isNuxtError(data) && data.toJSON(),```
packages/nuxt/src/app/plugins/revive-payload.client.ts
```NuxtError: data => createError(data)```

**Comment by sirgeon** [comment]
This behavior can be considered as part of the intended design, due to how serialization and deserialization work. However, the documentation may not explicitly detail this aspect, leading to confusion.

Heres a little work around which does manual deserialization for you:

```ts
// In your `error.vue` component or error handler
export default {
  async setup(props) {
    const error = props.error;

    // Check if error.data is a JSON string and parse it
    if (typeof error.data === 'string') {
      try {
        error.data = JSON.parse(error.data);
      } catch (e) {
        // Handle parsing error if necessary
      }
    }

    // Now error.data should be in its original form
    return { error };
  }
}
```
